### ISSUE TITLE
`beforeEach` function runs for every test despite not matching `grep`

### ISSUE THREAD

**Issue by darkwing** [theintern/intern#298]
During my functional testing, I created a suite with 6 tests and a `beforeEach` to set up each test.  I used the `grep` command line argument to identify a test I wanted to rewrite, which does identify just the one test correctly, but I noticed that `beforeEach` was executing 6 times, despite only the one test being executed.

This could still be intended but I thought I'd report it in case it was a bug; it could have unintended side-effects when only looking to run select tests with `grep`.

**Comment by johntrandall** [comment]
my $0.02: I found it very counter-intuitive that before/afterEach was run on skipped or grep-flitered tests. 

I do understand the runtime-logic as Colin explained it... but I can't get past the language. The assumption is that 'before each' is 'before each test that is run'. Skipped tests aren't really tests in my mind, they are placeholders. 

I've reverted to the somewhat messier method of commenting out tests that I don't want run, because I don't like the time that the extra 'beforeEach' commands add to the suite total-running-time.
